File: BUDESCU_BOGDAN/Proiect/backend/PsscProject/Models/Customers/Customer.cs
using PsscProject.Helpers.Domain;
using PsscProject.Models.Countries;
using PsscProject.Models.Generic;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Threading.Tasks;

namespace PsscProject.Models.Customers
{
    public class Customer : IAggregateRoot
    {
        private List<CreditCard> creditCards = new List<CreditCard>();
        public Guid Id { get; set; }
        public PlainText FirstName { get; set; }
        public PlainText LastName { get; set; }
        public Email Email { get; set; }
        public PlainText Password { get; set; }
        public Balance Balance { get; set; }
        public Address Address { get; set; }
        public Guid CountryId { get; set; }
        public ReadOnlyCollection<CreditCard> CreditCards { get { return this.creditCards.AsReadOnly(); } }

        public static Customer Create(PlainText firstname, PlainText lastname, Email email, Country country)
        {
            Customer customer = new Customer()
            {
                Id = Guid.NewGuid(),
                FirstName = firstname,
                LastName = lastname,
                Email = email,
                CountryId = country.Id
            };

            DomainEvents.Raise<CustomerCreated>(new CustomerCreated(customer));

            return customer;
        }

        public ReadOnlyCollection<CreditCard> GetCreditCardsAvailble()
        {
            return this.creditCards.FindAll(creditCard => creditCard.Active && creditCard.Expiry >= DateTime.Today).AsReadOnly();
        }

        public void AddCreditCard(CreditCard creditCard)
        {
            this.creditCards.Add(creditCard);
        }
    }
}

File: BUDESCU_BOGDAN/Proiect/frontend/src/app/models/Customer.ts
export class Customer {
    public id: string;
    public firstName: string;
    public lastName: string;
    public email: string;
    public countryId: string;
    public userName: string;
    public password: string;
    public street: string;
    public zipCode: string;
    public city: string;
    public image: string;
    public user : {
        username: string,
        password:string,
    }
    constructor(id: string, firstName: string, lastName: string, email: string, countryId: string, userName: string, password: string, street: string, zipCode: string, city: string, image: string) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.countryId = countryId;
        this.userName = userName;
        this.password = password;
        this.street = street;
        this.zipCode = zipCode;
        this.city = city;
        this.image = image;
    }
}
